get lastBackup fails when no backup is present in apiserver

[getLastBackup](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/rel-2018.T08a/api-controllers/ServiceFabrikApiController.js#L355) call fails with `“Unhandled error: AssertionError: Property ‘resourceId’ is required to get resource” ` if deployment resource is not tagged with last backup guid. ([ref](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/rel-2018.T08a/api-controllers/ServiceFabrikApiController.js#L365))

--- a/api-controllers/ServiceFabrikApiController.js
+++ b/api-controllers/ServiceFabrikApiController.js
@@ -23,6 +23,7 @@ const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
 const JsonWebTokenError = jwt.JsonWebTokenError;
 const ContinueWithNext = errors.ContinueWithNext;
 const DeploymentAlreadyLocked = errors.DeploymentAlreadyLocked;
+const AssertionError = assert.AssertionError;
 const ScheduleManager = require('../jobs');
 const config = require('../common/config');
 const CONST = require('../common/constants');
@@ -369,7 +370,7 @@ class ServiceFabrikApiController extends FabrikBaseController {
           resourceId: backupGuid
         })
       )
-      .catch(NotFound, (err) => {
+      .catch(NotFound, AssertionError, err => {
         // This code block is specifically for the transition of Service Fabrik to v2
         // Here we reffer to BackupService to get the lastBackup status
         logger.info('Backup metadata not found in apiserver, checking blobstore. Error message:', err.message);

--- a/data-access-layer/eventmesh/ApiServerClient.js
+++ b/data-access-layer/eventmesh/ApiServerClient.js
@@ -338,13 +338,13 @@ class ApiServerClient {
     assert.ok(opts.metadata || opts.options || opts.status, `Property 'metadata' or 'options' or 'status' is required to patch resource`);
     return this.getResource(opts)
       .then(resource => {
-        if (opts.status && opts.status.response && resource.status) {
-          const oldResponse = resource.status.response;
+        if (_.get(opts, 'status.response') && resource.status) {
+          const oldResponse = _.get(resource, 'status.response');
           const response = _.merge(oldResponse, opts.status.response);
           _.set(opts.status, 'response', response);
         }
         if (opts.options && resource.spec) {
-          const oldOptions = resource.spec.options;
+          const oldOptions = _.get(resource, 'spec.options');
           const options = _.merge(oldOptions, opts.options);
           _.set(opts, 'options', options);
         }
@@ -452,7 +452,7 @@ class ApiServerClient {
       .omit('operationName', 'operationType')
       .value();
     return this.getResource(options)
-      .then(json => json.metadata.labels[`last_${opts.operationName}_${opts.operationType}`]);
+      .then(json => _.get(json.metadata, `labels.last_${opts.operationName}_${opts.operationType}`));
   }
 
   /**
@@ -463,7 +463,7 @@ class ApiServerClient {
    */
   getOptions(opts) {
     return this.getResource(opts)
-      .then(resource => resource.spec.options);
+      .then(resource => _.get(resource, 'spec.options'));
   }
 
   /**
@@ -474,7 +474,7 @@ class ApiServerClient {
    */
   getResponse(opts) {
     return this.getResource(opts)
-      .then(resource => resource.status.response);
+      .then(resource => _.get(resource, 'status.response'));
   }
 
   /**
@@ -485,7 +485,7 @@ class ApiServerClient {
    */
   getResourceState(opts) {
     return this.getResource(opts)
-      .then(resource => resource.status.state);
+      .then(resource => _.get(resource, 'status.state'));
   }
 
   /**
@@ -496,7 +496,7 @@ class ApiServerClient {
    */
   getLastOperation(opts) {
     return this.getResource(opts)
-      .then(resource => resource.status.lastOperation);
+      .then(resource => _.get(resource, 'status.lastOperation'));
   }
 
 }

--- a/test/test_broker/acceptance/service-fabrik-api-2.0.instances.director.spec.js
+++ b/test/test_broker/acceptance/service-fabrik-api-2.0.instances.director.spec.js
@@ -676,6 +676,31 @@ describe('service-fabrik-api-sf2.0', function () {
             });
         });
 
+        it('should return 200 Ok - should check blobstore if last backup label is not set in deployment resource', function () {
+          mocks.uaa.tokenKey();
+          mocks.cloudController.findServicePlan(instance_id, plan_id, 2);
+          mocks.cloudController.getSpaceDevelopers(space_guid);
+          mocks.apiServerEventMesh.nockGetResource('deployment', 'director', instance_id, {});
+          const backupPrefix = `${space_guid}/backup/${service_id}.${instance_id}`;
+          const backupFilename = `${space_guid}/backup/${service_id}.${instance_id}.${backup_guid}.${started_at}.json`;
+          mocks.cloudProvider.list(container, backupPrefix, [backupFilename]);
+          const pathname = `/${container}/${backupFilename}`;
+          mocks.cloudProvider.download(pathname, data);
+          // mocks.agent.lastBackupOperation(backupState);
+          return chai.request(apps.external)
+            .get(`${base_url}/service_instances/${instance_id}/backup`)
+            .set('Authorization', authHeader)
+            .query({
+              space_guid: space_guid,
+            })
+            .catch(err => err.response)
+            .then(res => {
+              expect(res).to.have.status(200);
+              expect(res.body).to.eql(_.omit(data, 'agent_ip'));
+              mocks.verify();
+            });
+        });
+
         it('should return 404 if Not Found in blobstore and apiserver', function () {
           const backupPrefix = `${space_guid}/backup/${service_id}.${instance_id}`;
           const backupFilename = `${space_guid}/backup/${service_id}.${instance_id}.${backup_guid}.${started_at}.json`;

